sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/AutoCheckpointer.java

```diff
@@ -58,10 +65,15 @@ public void close(ChangeFeedObserverContext context, ChangeFeedObserverCloseReas
 
         if (this.isCheckpointNeeded()) {
             return context.checkpoint()
-                .doOnSuccess((Void) -> {
+                .onErrorResume(throwable -> {
+                    logger.warn("Checkpoint failed; this worker will be killed", throwable);
```

Comment:
logger.error ?
logger.debug ?

Reply:
Same as above, the exception could be thrown from the user code which handles the feeds. This is more or less for tracking purposes, as we want to capture as part of the default logs any case where we might have failed to check-point.